micov/_quant.py

```diff
@@ -0,0 +1,102 @@
+import numpy as np
+import polars as pl
+
+
+def make_csv_ready(df):
+    return df.with_columns(
+        (
+            pl.lit("[") +
+            pl.col(x)
+            .list.eval(pl.element().cast(str))
+            .list.join(",") +
+            pl.lit("]")
+        ).alias(x)
+        for x, y in df.schema.items()
+        if y == pl.List(pl.String) or y == pl.List(pl.Int64)
+    )
+
+
+def create_bin_list(genome_length, bin_num):
+    # note that bin_list is adjusted to 1-indexed to be compatible with pl.cut
+    bin_list_pos_stop = (
+        pl.Series("a", [0, genome_length], strict=False)
+        .hist(bin_count=bin_num)
+        .lazy()
+        .select(pl.col("breakpoint").alias("bin_stop"))
+        .with_row_index("bin_idx", offset=1)
+    )
+    bin_list_pos_start = (
+        pl.Series("a", [0, genome_length], strict=False)
+        .hist(bin_count=bin_num)
+        .lazy()
+        .select(pl.col("breakpoint").alias("bin_start"))
+        .with_row_index("bin_idx", offset=2)
+    )
+    bin_list = (
+        bin_list_pos_start.join(bin_list_pos_stop, on="bin_idx", how="right")
+        .fill_null(0)
+        .select([pl.col("bin_idx"), pl.col("bin_start"), pl.col("bin_stop")])
+        .with_columns(pl.col("bin_idx").cast(pl.Int64))
+        .collect()
```

Comment:
It looks like on line 47 that this result is cast to `lazy` anyway. Is it necessary to materialize here?